src/ros/thread/BagToImagesThread.cpp

```diff
@@ -128,16 +141,16 @@ BagToImagesThread::run()
     // Writing images might take lots of time, especially if higher compression is used. Thus, we aim to multithread the image writing.
     // However, the reader does not support parallel reading, thus we store the messages in a queue for parallel access.
     // Messages have a pretty large size and storing all of them at once might lead to an overflow quickly, though.
-    // Thus, iterate through the bag in steps of 100 messages:
-    // 1. Read 100 messages into the queue
-    // 2. Write messages to images, emptying the queue
-    // 3. Repeat until done
+    // Thus, we use a central queue storing up to 100 messages. One thread constantly writes messages into the queue,
+    // while the other threads will read messages out of and remove them from the queue.
     while (reader.has_next()) {
         if (isInterruptionRequested()) {
             return;
         }
 
-        readMessagesToQueue();
+        auto readingThread = std::thread([&pushMessagesToQueue] {
+            pushMessagesToQueue();
+        });
 
         std::vector<std::thread> threadPool;
         for (unsigned int i = 0; i < m_numberOfThreads; ++i) {
```

Comment:
Reduce the number of threads used here by one -> one thread for reading, everything else for writing